opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
+        - A FieldMetaInformation object (note that this contains obsolete
+            information about the transversal coordinate in 1D case)
+        """
+        # Check if polarization has been entered
+        if pol is None:
+            raise ValueError('The `pol` argument is missing or erroneous.')
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Test that polarization is in x or y plane for 3D cart. coordinates
+        if self.geometry == "3dcartesian" or self.geometry == "2dcartesian":
+            if pol == 0:
+                slicing_dir = 'y'
+                coord = 'x'
+                pol = None
+            elif pol == np.pi/2:
+                slicing_dir = 'x'
+                coord = 'y'
+                pol = None
+            else:
+                raise ValueError('Only polarization in the x or y plane is '
+                                 'supported for carthesian coordinates')
+        else:
+            # Extract radial if in thetaMode
+            coord = 'r'
+            slicing_dir = None
+        # Get field data
+        field = self.get_field( t=t, iteration=iteration, field='E',
+                                coord=coord, theta=pol, m=m,
+                                slicing_dir=slicing_dir )
+        extent = field[1]
+        if index == 'center':
+            # Get central slice
+            field_slice = field[0][int( field[0].shape[0] / 2), :]
+            # Calculate inverse FFT of filtered FFT array
+            envelope = self._fft_filter(field_slice, freq_filter)
+        elif index == 'all':
+            envelope = [ self._fft_filter(field[0][i, :], freq_filter)
+                         for i in range(field[0].shape[0]) ]
+        else:
+            field_slice = field[0][index, :]
+            # Calculate inverse FFT of filtered FFT array
+            envelope = self._fft_filter(field_slice, freq_filter)
+        # Return the result
+        return( envelope, extent )
+
+    def _fft_filter(self, field, freq_filter):
+        """
+        Filters out high frequencies in input data. Frequencies higher than
+        freq_filter / 100 times the dominant frequency will be filtered.
+
+        Parameters
+        ----------
+        field : 1D array
+            Array with input data in time/space domain
+
+        freq_filter : float
+            Frequency range in percent around the dominant frequency which will
+            not be filtered out
+
+        Returns
+        -------
+        A 1D array with filtered input data in time/space domain
+        """
+        # Number of sample points
+        N = field.size
+        # Fourier transform of the field slice
+        fft_field_slice = np.fft.fft(field)
+        fft_freqs = np.fft.fftfreq(N)
+        # Find central frequency
+        central_freq_i = np.argmax(fft_field_slice[:N / 2])
```

Comment:
I am not sure about the following but wouldn't it make more physical sense to use the maximum of the absolute value
```python
central_freq_i = np.argmax( abs(fft_field_slice[:N / 2]) )
```
instead of the maximum of the real part
```python
central_freq_i = np.argmax( fft_field_slice[:N / 2] ) 
```
I guess I am a bit afraid that the modulations in the spectral phase could in principle lead the real part to be zero at the central frequency.